ref/test/CMakeLists.txt

```diff
@@ -63,6 +63,9 @@ ${PROJECT_SOURCE_DIR}/test/src/interpolateTest.f90
 )
 add_executable (interpolateTest ${test_src_files}  ${PROJECT_SOURCE_DIR}/src/interpolate.f90)
 
+if ( GPTL_FOUND )
+   target_link_libraries( ${PROJECT_NAME} PUBLIC GPTL::GPTL )
```

Comment:
I think you need to replace `${PROJECT_NAME}` with `interpolateTest` here.  Sorry I wasn't clear that the name of the target needed to be the name of the executable in question, which in this case is `interpolateTest`.